File: 605-can-place-flowers/can-place-flowers.java
class Solution {

    public boolean canPlaceFlowers(int[] flowerbed, int n) {

    int maxZero,start,end;
    start = end = -1;
    maxZero = (flowerbed.length == 1 && flowerbed[0] == 0) ? 1 : 0;

    for (int i = 0 ; i<flowerbed.length ;i++){

        if (flowerbed[i] == 0 && start == -1){
            start = i;  
        }
        if (flowerbed[i] == 1 && end  == -1 && start > -1){
            end = i;
        }
        if (start  >-1 && end  > -1){
            int diff = end - start;
            maxZero += (start == 0) ? (diff)/2 : ((diff) > 4 ? (diff)/3 +1 : (diff)/3);
            start = end = -1; 
        }
      
    }

    if(start > -1 && end == -1){
      end = flowerbed.length;
      maxZero +=  (start > 0) ? (end - start)/2 : (end + 1) /2;
    }
    return maxZero >= n ;
    
    }
}

File: 605-can-place-flowers/can-place-flowers.go
func canPlaceFlowers(flowerbed []int, n int) bool {
    if n == 0 {
		return true
	}
	if len(flowerbed) == 1 && flowerbed[0] == 0 {
		n--
		return n == 0
	}
	if len(flowerbed) == 1 && flowerbed[0] == 1 {
		return false
	}
	start, end := 0, 1

	for end < len(flowerbed) {

		if flowerbed[start] == 0 && flowerbed[end] == 0 {
			flowerbed[start] = 1
			n--
		} else if flowerbed[start] == 0 && flowerbed[end] == 1 {
			start++
			end++
		}
		start += 2
		end += 2
	}
	if start < len(flowerbed) && flowerbed[start] == 0 && flowerbed[start-1] == 0 {
		n--
	}
	return n <= 0
}